src/input.rs

```diff
@@ -145,17 +133,22 @@ where
             None => panic!("no value set for {:?}({:?})", Q::default(), key),
         };
 
-        let StampedValue { value, changed_at } = slot.stamped_value.read().clone();
+        let StampedValue {
+            value,
+            durability,
+            changed_at,
+        } = slot.stamped_value.read().clone();
 
-        db.salsa_runtime().report_query_read(slot, changed_at);
+        db.salsa_runtime()
+            .report_query_read(slot, durability, changed_at);
 
         Ok(value)
     }
 
-    fn is_constant(&self, _db: &DB, key: &Q::Key) -> bool {
+    fn durability(&self, _db: &DB, key: &Q::Key) -> Durability {
         self.slot(key)
-            .map(|slot| slot.stamped_value.read().changed_at.is_constant)
-            .unwrap_or(false)
+            .map(|slot| slot.stamped_value.read().durability)
+            .unwrap_or(Durability::LOW)
```

Comment:
why would we ask a durability of a non-existing key?

Reply:
That's really just the method for the debug trait that lets you access the durability of a key. I suppose it could panic if the key is not defined.

Reply:
made it panic